File: conversion/Examples/src/ast_v2/convert_rust_to_ts.ts
export function convert_rust_file_to_ts(path: unknown): Result {
  // Delegate to the AST-based pipeline used in the TS self-example:
  // Rust source at `path` → Module (from_rust_module) → tag → TS string.
  const module: Module = from_rust_module(path as any);
  tag_special_functions_for_path(module, path as any);
  return Ok(module_to_ts(module));
}

File: src/ast_v2/convert_rust_to_ts.rs
use std::path::Path;

use crate::ast_v2::ast::Module;
use crate::ast_v2::{from_rust_module, module_to_ts};

use crate::ast_v2::tagging::tag_special_functions_for_path;

pub fn convert_rust_file_to_ts(path: &Path) -> Result<String, String> {
    let mut module: Module = from_rust_module(path)?;
    tag_special_functions_for_path(&mut module, path);
    Ok(module_to_ts(&module))
}
